### PR TITLE
Combine DQ scales and multiply into DQ scales.

### PR DESCRIPTION
**Pull request by luweizhou2016**
### Details:
 - *INT8 tiny-SWIN model has the pattern of FC + Multiply-per channel + Multiply DQ multiply , which is not supported by current DQ API. Need to support multiple multiply into DQ scales.*

### Tickets:
 - *CVS-109328*

### CODE REVIEW

**Comment by maxnick** [comment]
@usstq , could you please review?

**Comment by dmitry-gorokhov** [comment]
@luweizhou2016 Is the PR ready for review?

**Comment by luweizhou2016** [comment]
> @luweizhou2016 Is the PR ready for review?

Yes. Ready for review.

**Comment by yuxu42** [comment]
@usstq Could you please review? Thanks!

File: src/plugins/intel_cpu/src/node.cpp [hunk 1]
```diff
@@ -1630,15 +1630,19 @@ void Node::addSupportedPrimDesc(const std::vector<PortConfigurator>& inPortConfi
...
+    if (DQScales.empty())
+        DQScales.resize(scaleSize, 1.0);
+    if (!(scaleSize == 1 || DQScales.size() == 1 || DQScales.size() == scaleSize))
+        IE_THROW() << "set invalid scales size , DQScales vector size: " << DQScales.size()
+                    << ", scale data size: " << scaleSize
+                    << "Node: ##" << getName();
+    if (scaleSize > DQScales.size())
+        DQScales.resize(scaleSize, DQScales[0]);
+    for (size_t i = 0; i < DQScales.size(); i++) {
+        DQScales[i] = DQScales[i] * (scaleSize == 1 ? scaleData[0] : scaleData[i]);
```
Comment (usstq) [inline src/plugins/intel_cpu/src/node.cpp:1644]: So `setDQScales` is actually merging new scales into existing ones? the function name is a little misleading to me.

File: src/plugins/intel_cpu/src/graph_optimizer.cpp [hunk 2]
```diff
@@ -260,7 +260,7 @@ void GraphOptimizer::FuseConvMatmulFCDeconvAndDQScales(Graph &graph) {
         auto scalesDims = getNormalizedDimsBySize(scales->getOutputShapeAtPort(0).getDims(),
                                                 node->getOutputShapeAtPort(0).getDims().size());
         auto scaleSize = std::accumulate(scalesDims.begin(), scalesDims.end(), 1, std::multiplies<size_t>());
-        node->initializeDQScales(scalesData, scaleSize);
+        node->setDQScales(scalesData, scaleSize);
```
Comment (usstq) [inline src/plugins/intel_cpu/src/graph_optimizer.cpp:263]: So `initializeDeQuantizedScales` will be also called twice in INT8 tiny-SWIN model on pattern of FC + Multiply-per channel + Multiply DQ multiply? the name `initialize` seems to imply that it's only called once for each node, maybe change it to `append` or `fuse`?

Comment (luweizhou2016) [reply src/plugins/intel_cpu/src/graph_optimizer.cpp]: rename to fuseDequantizedScale.

**Comment by luweizhou2016** [comment]
@usstq  Applied.

**Comment by dmitry-gorokhov** [comment]
@usstq Please finalize the review

**Comment by luweizhou2016** [comment]
@maxnick @dmitry-gorokhov , Pls also help to ensure  https://github.com/openvinotoolkit/openvino/pull/17082 is merged before this PR.  Otherwise maybe has potential issue in very little chance. Thx！

**Review by usstq — APPROVED** [review]
LGTM

**Comment by maxnick** [comment]
@luweizhou2016  please rebase the PR on latest master.

**Comment by maxnick** [comment]
@v-Golubev, @eshoguli could you please review the LPT part?

File: src/plugins/intel_cpu/src/node.cpp [hunk 3]
```diff
@@ -1630,15 +1630,19 @@ void Node::addSupportedPrimDesc(const std::vector<PortConfigurator>& inPortConfi
...
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/node.cpp:1644]:
```suggestion
    if (1 == scaleSize) {
        std::transform(DQScales.begin(), DQScales.end(),  DQScales.begin(), [=](float val){ return val * DQScales[0]; })
     } else {
         for (size_t i = 0; i < DQScales.size(); i++) {
             DQScales[i] *= scaleData[i];
         }
     }
```

File: src/plugins/intel_cpu/src/nodes/fullyconnected.cpp [hunk 4]
```diff
@@ -355,17 +356,20 @@ void FullyConnected::prepareParams() {
...
 
-        if (!found)
-            return nullptr;
+        if (found)
+            return std::make_shared<DnnlExecutor>(prim_desc);
 
-        return std::make_shared<DnnlExecutor>(prim_desc);;
+        // For dynamic shape, the expected implement type kernel can support with dummy shape but
+        // not support with run time inference shape. With this case, the implement type would be
+        // ignored and first available primitive descriptor would be chosen.
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/nodes/fullyconnected.cpp:371]:
```suggestion
        // For dynamic shape, the expected implement type kernel can support the dummy shape but
        // not the run time inference shape. In this case, the implementation type will be
        // ignored and the first available primitive descriptor will be chosen.
```

File: src/plugins/intel_cpu/tests/functional/subgraph_tests/src/fq_layer_dq_bias.cpp [hunk 5]
```diff
@@ -100,7 +103,8 @@ const std::vector<std::string> layer_types_4D_static = {
 
 INSTANTIATE_TEST_SUITE_P(smoke_FQLayerDQBias_4D_static, FQLayerDQBias,
                          ::testing::Combine(::testing::ValuesIn(input_shapes_4D_static),
-                                            ::testing::ValuesIn(layer_types_4D_static)),
+                                            ::testing::ValuesIn(layer_types_4D_static),
+                                            ::testing::ValuesIn({false})),
```
Comment (maxnick) [inline src/plugins/intel_cpu/tests/functional/subgraph_tests/src/fq_layer_dq_bias.cpp:107]:
```suggestion
                                            ::testing::Values(false)),
```
Here and below.

**Comment by luweizhou2016** [comment]
@maxnick , applied review comments.

File: src/plugins/intel_cpu/tests/functional/subgraph_tests/src/fq_layer_dq_bias.cpp [hunk 6]
```diff
@@ -111,7 +115,8 @@ const std::vector<std::string> layer_types_4D_dynamic = {
 
 INSTANTIATE_TEST_SUITE_P(smoke_FQLayerDQBias_4D_dynamic, FQLayerDQBias,
                          ::testing::Combine(::testing::ValuesIn(input_shapes_4D_dynamic),
-                                            ::testing::ValuesIn(layer_types_4D_dynamic)),
+                                            ::testing::ValuesIn(layer_types_4D_dynamic),
+                                            ::testing::ValuesIn({false})),
```
Comment (v-Golubev) [inline src/plugins/intel_cpu/tests/functional/subgraph_tests/src/fq_layer_dq_bias.cpp:119]:
```suggestion
                                            ::testing::Values(false)),
```

File: src/plugins/intel_cpu/tests/functional/subgraph_tests/src/fq_layer_dq_bias.cpp [hunk 7]
```diff
@@ -123,7 +128,14 @@ const std::vector<std::string> layer_types_2D = {
 
 INSTANTIATE_TEST_SUITE_P(smoke_FQLayerDQBias_2D, FQLayerDQBias,
                          ::testing::Combine(::testing::ValuesIn(input_shapes_2D),
-                                            ::testing::ValuesIn(layer_types_2D)),
+                                            ::testing::ValuesIn(layer_types_2D),
+                                            ::testing::ValuesIn({false})),
```
Comment (v-Golubev) [inline src/plugins/intel_cpu/tests/functional/subgraph_tests/src/fq_layer_dq_bias.cpp:132]:
```suggestion
                                            ::testing::Values(false)),
```

File: src/plugins/intel_cpu/tests/functional/subgraph_tests/src/fq_layer_dq_bias.cpp [hunk 8]
```diff
@@ -123,7 +128,14 @@ const std::vector<std::string> layer_types_2D = {
...
+                         FQLayerDQBias::getTestCaseName);
+
+INSTANTIATE_TEST_SUITE_P(smoke_FQLayerDQExtraMultiplyAdd_2D, FQLayerDQBias,
+                         ::testing::Combine(::testing::ValuesIn(input_shapes_2D),
+                                            ::testing::ValuesIn(layer_types_2D),
+                                            ::testing::ValuesIn({true})),
```
Comment (v-Golubev) [inline src/plugins/intel_cpu/tests/functional/subgraph_tests/src/fq_layer_dq_bias.cpp:138]:
```suggestion
                                            ::testing::Values(true)),
```

**Comment by luweizhou2016** [comment]
@v-Golubev , applied review comments.

**Review by v-Golubev — APPROVED** [review]
LPT part LGTM

**Review by maxnick — APPROVED** [review]
In general LGTM. It does make sense to run internal validation.

**Pull request merged**